onnx/defs/nn/defs.cc

```diff
@@ -2190,6 +2190,183 @@ ONNX_OPERATOR_SET_SCHEMA(
         }
         )ONNX"));
 
+void col2imShapeInference(InferenceContext& ctx) {
+  propagateElemTypeFromInputToOutput(ctx, 0, 0);
+
+  // All inputs shapes are required
+  if (!hasNInputShapes(ctx, 3)) {
+    return;
+  }
+
+  // We assume image_shape has correct spatial dimensions for next validations
+  // An alternative is get the the number of spatial dimensions as an input argument
+  Dim n_input_dims;
+  unifyInputDim(ctx, 1, 0, n_input_dims);
```

Comment:
I think we can also add the following check (since the next input (3rd input) should also have a matching dim):
```cc
unifyInputDim(ctx, 2, 0, n_input_dims);
```

Reply:
We should also ad the following checks to ensure that the ranks of both inputs are 1:
```cc
checkInputRank(ctx, 1, 1);
checkInputRank(ctx, 2, 1);
```